include/seqcmd.h

```diff
@@ -0,0 +1,510 @@
+    Audio_QueueSeqCmd((SEQCMD_OP_UNQUEUE_SEQUENCE << 28) | ((u8)(seqPlayerIndex) << 24) | \
+                      ((u8)(fadeOutInDuration) << 16) | (seqId))
+
+/**
+ * Set the volume of an entire sequence on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param duration the duration to set the volume over
+ * @param volume the target volume for the sequence
+ */
+#define SEQCMD_SET_PLAYER_VOLUME(seqPlayerIndex, duration, volume)                                              \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_PLAYER_VOLUME << 28) | ((u8)(seqPlayerIndex) << 24) | ((duration) << 16) | \
+                      (volume))
+
+/**
+ * Scale the frequency of an entire sequence on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param duration the duration to set the frequency over
+ * @param freq the scaling factor to shift the pitch, relative to 1000.
+ *
+ * @note 2000 will double the frequency (raise an octave), 500 will half the frequency (lower an octave)
+ */
+#define SEQCMD_SET_PLAYER_FREQ(seqPlayerIndex, duration, freq) \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_PLAYER_FREQ << 28) | ((u8)(seqPlayerIndex) << 24) | ((duration) << 16) | (freq))
+
+/**
+ * Scale the frequency of a specific channel on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param channelIndex the index of the channel to modify
+ * @param duration the duration to set the frequency over
+ * @param freq the scaling factor to shift the pitch, relative to 1000.
+ *
+ * @note a frequency of 2000 will double the frequency (raise an octave), 500 will half the frequency (lower an octave)
+ */
+#define SEQCMD_SET_CHANNEL_FREQ(seqPlayerIndex, channelIndex, duration, freq)                                  \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_CHANNEL_FREQ << 28) | ((u8)(seqPlayerIndex) << 24) | ((duration) << 16) | \
+                      ((channelIndex) << 12) | (freq))
+
+/**
+ * Set the volume of a specific channel on a given seqPlayer over a specified duration
+ *
+ * @param seqPlayerIndex the index of the seqPlayer to modify
+ * @param channelIndex the index of the channel to modify
+ * @param duration the duration to set the volume over
+ * @param volume the target volume for the sequence
+ */
+#define SEQCMD_SET_CHANNEL_VOLUME(seqPlayerIndex, channelIndex, duration, volume)                                    \
+    Audio_QueueSeqCmd((SEQCMD_OP_SET_CHANNEL_VOLUME << 28) | ((u8)(seqPlayerIndex) << 24) | ((u8)(duration) << 16) | \
+                      ((u8)(channelIndex) << 8) | ((u8)volume))
+
+/**
+ * Write a value to an array that can be read as input directly by the sequence itself. This will be set to the global
+ * ioPort, which can affect the entire sequence. How the sequence responds this io value will entirely depend
```

Comment:
in general feel like the fact that its an array is not relevant to its function or purpose. same goes for the other io descriptions

Reply:
also "how the sequence responds this" (was copy pasted to a few places)

Reply:
yeah "to an array" can just be omitted
The note describing io ports are 0-7 is enough